test/app/exercises_test.rb

```diff
@@ -48,4 +48,15 @@ def test_bulk_delete
     post "/exercises/delete", { exercise_ids: [exercise.id, exercise_2.id] }, login(alice)
     assert_equal 0, UserExercise.count
   end
+
+  def test_request_help_for_exercise
+    post "/exercises/#{exercise.key}/request_help", {}, login(alice)
+    assert exercise.reload.request_help
+  end
+
+  def test_remove_help_for_exercise
+    post "/exercises/#{exercise.key}/remove_help", {}, login(alice)
+    refute exercise.reload.request_help
```

Comment:
Keep in mind that you'll be getting a brand new exercise object in each test, independent from the other tests.

This means that `excercise.request_help` will _already_ be `false`, even if you don't call the `post` to the `remove_help` route. You can try that out by commenting out the previous line. If the test still passes, as I expect it will, then it's not really testing the functionality.

To fix this, you have a few options:

1. You could manually set the `request_help` flag on `exercise` (and save it) before calling `post`
2. You could call both `request_help` and then `remove_help` in sequence
3. You could set `request_help` to `true` in `exercise_2` where that is constructed in `setup` and use that instead of `exercise` in this test.

Of these options, I personally prefer #1, since it makes obvious the connection between setting `request_help` and the behaviour in the test, but some people might prefer #3. #2 isn't as good, because then you'd be testing the behaviour of both routes in one test.

It's always a good idea when writing tests to try breaking the test to make sure it fails. This is how you know you have a good test. You can do this by commenting out parts of the production code and running the tests to make sure that they fail. An even better approach is to use test-driven development: write the test to fail _first_, and then write the code to make it pass.